R/qc_cvs.R

```diff
@@ -0,0 +1,174 @@
+#' @param plot_style Variable indicating the plotting style. \code{plot_style = "boxplot"} plots a boxplot, whereas \code{plot_style = "density"} plots the CV density distribution. \code{plot_style = "violin"} returns a violin plot. Default is \code{plot_style = "density"}.
+#'
+#' @return Either the median CVs in % or a plot showing the distribution of the CVs.
+#' @import dplyr
+#' @import ggplot2
+#' @importFrom tidyr drop_na
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+#' @importFrom stats sd
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' qc_cvs(
+#' data,
+#' grouping = pep_stripped_sequence,
+#' condition = condition,
+#' intensity = fg_quantity,
+#' plot = TRUE,
+#' plot_style = "density"
+#' )
+#' }
+qc_cvs <-
+  function(data, grouping, condition, intensity, plot = TRUE, plot_style = "density")
+  {
+    if(plot == FALSE)
+    {
+      if(max(dplyr::pull(data, {{intensity}})) < 1000)
+      {
+      stop("Please backtransform your data or use raw values. The function does not handle log2 transformed data.")
+      }
+
+      result <- data %>%
+        dplyr::distinct({{grouping}}, {{condition}}, {{intensity}}) %>%
+        tidyr::drop_na({{intensity}}) %>%
+        dplyr::group_by({{grouping}}) %>%
+        dplyr::mutate(cv_combined = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::group_by({{condition}}, {{grouping}}) %>%
+        dplyr::mutate(cv = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::distinct({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
+        tidyr::drop_na() %>%
+        dplyr::group_by({{condition}}) %>%
+        dplyr::mutate(median_cv = stats::median(.data$cv)) %>%
+        dplyr::ungroup() %>%
+        dplyr::mutate(median_cv_combined = stats::median(.data$cv_combined)) %>%
+        dplyr::select(- {{grouping}}, - .data$cv_combined, - .data$cv) %>%
+        dplyr::distinct()
+
+      return(result)
+    }
+
+    if(plot == TRUE)
+    {
+      if(max(dplyr::pull(data, {{intensity}})) < 1000)
+      {
+        stop("Please backtransform your data or use raw values. The function does not handle log2 transformed data.")
+      }
+      result <- data %>%
+        dplyr::distinct({{grouping}}, {{condition}}, {{intensity}}) %>%
+        tidyr::drop_na({{intensity}}) %>%
+        dplyr::group_by({{grouping}}) %>%
+        dplyr::mutate(cv_combined = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::group_by({{condition}}, {{grouping}}) %>%
+        dplyr::mutate(cv = (stats::sd({{intensity}}) / mean({{intensity}})) * 100) %>%
+        dplyr::ungroup() %>%
+        dplyr::distinct({{condition}}, {{grouping}}, .data$cv_combined, .data$cv) %>%
+        tidyr::drop_na() %>%
+        tidyr::pivot_longer(cols = starts_with("cv"), names_to = "type", values_to = "values") %>%
+        dplyr::mutate(type = ifelse(.data$type == "cv", {{condition}}, "combined")) %>%
```

Comment:
you can try to make it a factor and reorder it so that combined is first or last.